docs/source/development-guidelines.rst

```diff
@@ -0,0 +1,223 @@
+Zipline Development Guidelines
+==============================
+This page is intended for developers of Zipline, people who want to contribute to the Zipline codebase or documentation, or people who want to install from source and make local changes to their copy of Zipline.
+
+All contributions, bug reports, bug fixes, documentation improvements, enhancements and ideas are welcome. We `track issues`__ on `GitHub`__ and also have a `mailing list`__ where you can ask questions.
+
+__ https://github.com/quantopian/zipline/issues
+__ https://github.com/
+__ https://groups.google.com/forum/#!forum/zipline
+
+Creating a Development Environment
+----------------------------------
+
+First, you'll need to clone Zipline by running:
+
+.. code-block:: bash
+
+   git clone git@github.com:your-github-username/zipline.git
+
+Then check out to a new branch where you can make your changes:
+
+.. code-block:: bash
+		
+   git checkout -b some-short-descriptive-name
+
+The following section assumes you already have virtualenvwrapper and pip installed on your system. If you don't already have them, you'll need some C library dependencies. You can follow the `install guide`__ to get the appropriate dependencies.
+
+__ http://www.zipline.io/install.html 
+
+Suggested installation of Python library dependencies used for development:
+
+.. code-block:: bash
+
+   $ mkvirtualenv Zipline
+   $ ./etc/ordered_pip.sh ./etc/requirements.txt
+   $ pip install -r ./etc/requirements_dev.txt
+   $ pip install -r ./etc/requirements_blaze.txt 
+   $ pip install -r ./etc/requirements_talib.txt 
+
+Finally, you can build the C extensions by running:
+
+.. code-block:: bash
+
+   $ python setup.py build_ext --inplace
+
+To finish, make sure tests pass by running
+
+.. code-block:: bash
+
+   $ nosetests
+
+If you get an error running nosetests after setting up a fresh virtualenv, please try running
+
+.. code-block:: bash
+
+   # where zipline is the name of your virtualenv
+   $ deactivate Zipline
+   $ workon Zipline
+
+
+Development with Docker
+-----------------------
+
+If you want to work with zipline using a `Docker`__ container, you'll need to build the ``Dockerfile`` in the Zipline root directory, and then build ``Dockerfile-dev``. Instructions for building both containers can be found in ``Dockerfile`` and ``Dockerfile-dev``, respectively.
+
+__ https://docs.docker.com/get-started/
+
+
+Style Guide & Running Tests
+---------------------------
+
+We use `flake8`__ for checking style requirements and `nosetests`__ to run Zipline tests. Our `continuous integration`__ tools will run these commands.
+
+__ http://flake8.pycqa.org/en/latest/
+__ http://nose.readthedocs.io/en/latest/
+__ https://en.wikipedia.org/wiki/Continuous_integration
+
+Before submitting patches or pull requests, please ensure that your changes pass when running:
+
+.. code-block:: bash
+
+   $ flake8 Zipline tests
```

Comment:
Especially for case-sensitive environments, this should probably be lowercase "zipline" for the directory.